google-cloud-pubsub/acceptance/pubsub/pubsub_test.rb

```diff
@@ -161,6 +162,7 @@ def retrieve_snapshot project, subscription, snapshot_name
       _(subscription.filter).must_be :frozen?
       _(subscription.retry_policy.minimum_backoff).must_equal retry_minimum_backoff
       _(subscription.retry_policy.maximum_backoff).must_equal retry_maximum_backoff
+      _(subscription.detached?).must_equal true
```

Comment:
Currently failing with:

```
  1) Failure:
Google::Cloud::PubSub::pubsub::Subscription on Topic#test_0002_should allow create, update and delete of detached subscription [/Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-pubsub/acceptance/pubsub/pubsub_test.rb:165]:
Expected: true
  Actual: false
```

Reply:
This does **NOT** pass currently with `endpoint: staging-pubsub.sandbox.googleapis.com`.

Reply:
What project are you using to test this?

Reply:
Sorry, I should have noted that. It's failing currently on both of these projects:

helical-zone-771 (CI)
bustling-kayak-91516  (individual development)

Reply:
You've uncovered a backend bug! I should prevent users from creating subscriptions with "detached" set to true. "detached" should only change after calling DetachSubscription. I'll fix it. It should be in staging by mid next week.

We should change this test to call DetachSubscription after creation, and then check if detached is true afterward.

Reply:
I guess it doesn't make much sense to create an already detached subscription, but fortunately for the QA effort I didn't pay any attention and added the option anyway.

I removed `detached` from `Topic#subscribe`.

System tests passing now against staging.